### PR TITLE
fix: better tenderly sim error handling

### LINKED ISSUE #800: feat: skip tenderly sims if the env vars are not present
this for devs that dont have the prod keys on their local

i made an attempt in #799 but sims were also skipped there in ci 🤔

### LINKED ISSUE #902: bug: Fix tenderly sims
In this RUN ID the reporting failed on tenderly sim:
https://github.com/BalancerMaxis/multisig-ops/actions/runs/8736442395/job/23971305867

In this PR I removed the tenderly sim functionatliy from the reports, and then ran again/everything worked.  I also successfully simmed one of the payloads in tenderly via safe:
https://github.com/BalancerMaxis/multisig-ops/pull/901. - see https://github.com/BalancerMaxis/multisig-ops/pull/901/commits/057b524b28378f859eba9ec982489c1c6d6058c3

Tenderly generation will be globally disabled once that BIP is merged.

### PR DESCRIPTION
**Pull request by gosuto-inzasheru**
closes #902 
closes #800

### CODE REVIEW

**Comment by gosuto-inzasheru** [comment]
- graceful catching of either having no tenderly env vars or running into issues with payload json interpretation
- solves the skipping in #903
- the convoluted nested tuple in #904 is not solved, will not spend time on that for now

File: .github/workflows/black.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
         uses: psf/black@stable
         with:
           options: "--verbose"
-          version: "24.4.0"
```
Comment (Tritium-VLK) [inline .github/workflows/black.yaml:16]: Why the change to black version here?  How does this relate to tenderly workflows?

**Review by Tritium-VLK — COMMENTED** [review]
If you've tested it and it works it looks fine to me.  Maybe run it on a bunch of moved files in 4269-W69?

Why the black version change?

File: .github/workflows/black.yaml (hunk 1 above)
Comment (Tritium-VLK) [reply .github/workflows/black.yaml:16]: Figured this out in my next review... #908

**Review by Tritium-VLK — APPROVED** [review]
